scripts/strun

```diff
@@ -24,20 +24,12 @@ if __name__ == '__main__':
 
     try:
         step = Step.from_cmdline(sys.argv[1:])
-
-    except NoDataOnDetectorError:
-        #  No science data is present on a detector. This can happen
-        #  with NIRSpec and the NRS2 detector: Situations occur when no spectra
-        #  disperse across both detectors.
-        #
-        #  Special exit status is provided so that automatic processing can detect
-        #  this situation and act accordingly.
-        #
-        #  Full discussion: https://github.com/spacetelescope/jwst/issues/2336
+    except SystemExit as e:
```

Comment:
`e` is not being used in this `except` block.

And since `SystemExit` subclasses `BaseException` instead of `Exception`, it's possible we don't even need this `except` block.  I.e. the `except Exception` block below won't catch `SystemExit`, so it will be raised from the step code regardless.

Reply:
Oh hey so it does!  Thanks, I'll fix that up.

Reply:
(now updated to define and catch an `StpipeExitException`)

Reply:
I guess the one advantage of capturing it here and printing the traceback is that normally when `SystemExit` is raised, it just exits, without printing the traceback.  At least that's the description here:

https://docs.python.org/3/library/exceptions.html#SystemExit

🤔

Reply:
I played around with this and noticed that catching `SystemExit` results in tracebacks when argparse prints its help output, since it raises `SystemExit` to bail out in that case.